### ISSUE TITLE
git with recursive=true always reports status as changed

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#30043]
##### ISSUE TYPE

Bug Report

##### COMPONENT NAME

git module

##### ANSIBLE VERSION

2.1

##### SUMMARY

From @LinusU on 2015-09-16T15:19:36Z
The git module always reports `changed` when run with `recursive=true`, even if there are no new commits at all.

It works great when running without `recursive=true` thought :+1: 

Copied from original issue: ansible/ansible-modules-core#2094

**Comment by ansibot** [comment]
From @w1r0x on 2015-09-16T15:19:36Z

Do you use track_submodules=yes? If so it can be related to #1509

**Comment by ansibot** [comment]
From @LinusU on 2015-09-16T15:19:36Z

I do not use `track_submodules` so it's probably another issue...

**Comment by ansibot** [comment]
From @robinro on 2015-09-16T15:19:36Z

@LinusU 
Can you provide a minimal working example to reproduce this? i.e. example playbook and repo?
`recursive=true` is not the only necessary condition (gives changed=false for me).

**Comment by ansibot** [comment]
From @jctanner on 2015-09-16T15:19:36Z

@LinusU any response to @robinro's question?

needs_info

**Comment by ansibot** [comment]
From @ansibot on 2015-09-16T15:19:36Z

@LinusU ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @ansibot on 2015-09-16T15:19:36Z

@LinusU ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @ansibot on 2015-09-16T15:19:36Z

@LinusU ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @ansibot on 2015-09-16T15:19:36Z

@LinusU ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansibot Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- issue type
- ansible version
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (112/100 (112%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext8.iad.github.net ([192.30.252.199]:34848 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1drnEY-000iVi-M2
	for greg@biomassiv.es; Tue, 12 Sep 2017 10:36:18 -0500
Date: Tue, 12 Sep 2017 08:28:46 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1505230538;
	bh=O9riGBJC6vic+b0TNeEQ9FqzKgLam3PmdaZXg88Bj38=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=m1IjXhAUtkk/zU/YAedWlJt8vCaTgc/EZy9vKSFjlGbJ4lkMqjJSIpwI2bkyFDNyQ
	 H4VRSzDAwvRUEPFZ34r0H6cSBS+qzWGx8KBPl/MoJ+YG9fjfBlNTvDabHS+jz0RQ5O
	 B6nbll2KU/yNQxJufab6jsFo0ZQ6IExTsZRoKdRU=

**Comment by Shrews** [comment]
Cannot reproduce this on current versions (>= 2.9). Since no reproducer example was given, going to close this. If you can reproduce the behavior with latest versions, please open a new bug and provide a simple example that recreates the bug.
